services/src/Microsoft.Azure.IIoT.Services.Processor.Events/src/Program.cs

```diff
@@ -128,6 +128,11 @@ IConfiguration configuration
             builder.RegisterType<EventProcessorFactory>()
                 .AsImplementedInterfaces();
 
+            // Prometheus metric server
+            builder.RegisterType<MetricServerHost>()
+                .AsImplementedInterfaces().SingleInstance()
+                .WithParameter("port", 9500);
```

Comment:
This is brittle, the parameter name must now never change in the "implementation".   A new implementation must use exactly the same name for the parameter, etc.   Please add interface or use an existing interface to define the metricsserver port on the Config.cs class which is already injected.    See below for the other occurrences.